pkg/indexer/common/log_handler.go

```diff
@@ -167,21 +168,32 @@ func IndexLogs(
 }
 
 func retry(
+	ctx context.Context,
 	logger *zap.Logger,
 	sleep time.Duration,
 	address string,
 	fn func() re.RetryableError,
 ) error {
 	for {
-		if err := fn(); err != nil {
-			logger.Error("error storing log", zap.Error(err))
-			if err.ShouldRetry() {
-				metrics.EmitIndexerRetryableStorageError(address)
-				time.Sleep(sleep)
-				continue
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if err := fn(); err != nil {
+				logger.Error("error storing log", zap.Error(err))
+				if err.ShouldRetry() {
+					metrics.EmitIndexerRetryableStorageError(address)
+
+					select {
+					case <-ctx.Done():
+						return ctx.Err()
+					case <-time.After(sleep):
```

Comment:
@coderabbitai can you check whether time.after works in a loop? Can you show us the relevant docs.